### PR TITLE
Move GPS logging into AP_GPS

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS_Logging.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "AP_GPS.h"
+
+#include <AP_Logger/AP_Logger.h>
+
+// Write an GPS packet
+void AP_GPS::Write_GPS(uint8_t i)
+{
+    const AP_GPS &gps = AP::gps();
```
Comment (hendjoshsr71) [inline libraries/AP_GPS/AP_GPS_Logging.cpp:8]: Could remove this and all the `gps.` below.

```suggestion
```

**Comment by hendjoshsr71** [comment]
Could also remove https://github.com/ArduPilot/ardupilot/blob/8058e8de7d114c68a8637e7c7ce246c4dca378d9/libraries/AP_Logger/LogFile.cpp#L9

File: libraries/AP_GPS/AP_GPS_Logging.cpp (hunk 1 above)
Comment (peterbarker) [reply libraries/AP_GPS/AP_GPS_Logging.cpp]: Done, thanks.

It's possible @WickedShell might bounce this separate patch as I'm no longer just moving code around.

**Comment by peterbarker** [comment]
As requested by @WickedShell , I've moved the logging functions from `AP_GPS_Logging.cpp` back into `AP_GPS.cpp`

**Comment by tridge** [comment]
need to test blended GPS logging works

**Comment by peterbarker** [comment]
Blended GPS logging does work, at least in-as-far as the data in in `GPS[2].Lat` looks reasonable.

**Pull request merged**